App/Logic/SettingsLogic.swift

```diff
@@ -240,6 +240,77 @@ extension Logic.Settings {
   }
 }
 
+// MARK: Update Country
+
+extension Logic.Settings {
+  /// Shows the flow to update the country
+  struct ShowUpdateCountry: AppSideEffect {
+    func sideEffect(_ context: SideEffectContext<AppState, AppDependencies>) throws {
+      let state = context.getState()
+
+      let dummyIngestionWindowDuration = state.configuration.dummyIngestionWindowDuration
+      let countries = state.exposureDetection.countriesOfInterest
+      let lan = Locale.current.languageCode ?? "en"
+      let countryList: [String: String] = state.configuration.countries[lan] ?? state.configuration.countries["en"]!
+
+      try context.awaitDispatch(Show(
+        Screen.updateCountry,
+        animated: true,
+        context: CountriesOfInterestLS(
```

Comment:
There is no need to pass that many information in the local state. The local state should be used to pass information that the VC alone cannot retrieve from the state. For instance, in a VC that shows a user profile, the id of the user can be something passed to the local state. In this case all the variables can be taken directly from the state in the view model